scripts/src/admin/translations.ts

```diff
@@ -0,0 +1,157 @@
+import { ExportToCsv, Options as ExportCsvOptions } from "export-to-csv";
+import { FindManyOptions } from "typeorm";
+
+import { writeFile } from "fs";
+
+import { ContentType, Offer, OfferContent } from "../models/offers";
+
+type CsvRow = {
+	Type: string;
+	Key: string;
+	Default: string;
+	Translation: "";
+	"Character limit": number;
+};
+
+type OfferContentContent = {  // I know, but I didn't have a better name for the 'content' field in the OfferContent table
+	pages?: any[]
+	confirmation?: {
+		title: string,
+		description: string,
+		image: string
+	}
+};
+
+const CHARACTER_LIMITS: { [path: string]: number } = {
+	"poll:offer:title": 14,
+	"poll:offer:description": 18,
+	"poll:offer:orderTitle": 8,
+	"poll:offer:orderDescription": 24,
+	"poll:offer_contents:content:pages.title": 38,
+	"poll:offer_contents:content:pages.description": 24,
+	"poll:offer_contents:content:pages.question.answers": 22,
+	"quiz:offer:title": 14,
+	"quiz:offer:description": 18,
+	"quiz:offer:orderTitle": 8,
+	"quiz:offer:orderDescription": 24,
+	"quiz:offer_contents:content:pages.description": 66,
+	"quiz:offer_contents:content:pages.question.answers": 22,
+};
+
+const KEY_TO_PATH_REGEX = /\b([\w_]+:)\w+:|\[\d\]/g;
+
+const EXCLUDED = [
+	"tutorial",
+	"quiz:offer_contents:content:pages.question.id",
+	"poll:offer_contents:content:pages.question.id",
+];
+
+function constructRow(contentType: ContentType, key: string, str: string) {
+	const path = `${contentType}:${key.replace(KEY_TO_PATH_REGEX, "$1")}`;
```

Comment:
can you give an example for the input and output of this line?

Reply:
input:
contentType: `poll`
key: `offer_contents:O45hzJEibwUdNCLXTlmtY:content:pages[3].description`
output: `poll:offer_contents:content:pages.description`

`poll:offer_contents:content:pages.description`

Reply:
@doodyparizada ^